### PR TITLE
[CI] Remove template explosion on the maui templates build tests.

### PR DESCRIPTION
**Pull request by mandel-macaque**
Move to minimize the size of the yaml by using a matrix over expanding the same templates twice. This will reduce the final size of the yaml of the pipeline.

### CODE REVIEW

File: eng/pipelines/common/maui-templates.yml [hunk 1]
```diff
@@ -55,69 +55,76 @@ parameters:
...
+    continueOnError: true
+
+  - pwsh: |
+      Write-Host "Current job status is: $env:AGENT_JOBSTATUS"
+      if ($env:AGENT_JOBSTATUS -eq "SucceededWithIssues") {
+          Write-Host "##vso[task.complete result=Failed;]DONE"
+      }
+    displayName: Fail if any issues occurred
 
 - ${{ each RunPlatform in parameters.RunPlatforms }}:
```
Comment (pjcollins) [inline eng/pipelines/common/maui-templates.yml:129]: Should we do the same for the run templates jobs? I think we'd want to use the matrix strategy here as well if it produces less YAML but that could be done in a future PR.

Comment (mandel-macaque) [reply eng/pipelines/common/maui-templates.yml:128]: You are 100% correct, I was going to dos it in a future PR :)

**Review by pjcollins — APPROVED** [review]

**Pull request merged**